http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,219 @@
+        assertNotNull(vertx);
+        assertNotNull(httpClient);
+        vertxTestContext
+                .verify(() -> {
+                    httpClient.request(HttpMethod.GET, app.getPort(Protocol.HTTPS), baseUri.getHost(), BASE_ENDPOINT)
+                            .compose(req -> req.send()
+                                    .compose(httpClientResponse -> {
+                                        requestCheckpoint.flag();
+                                        assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                                        assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                                        return httpClientResponse.body();
+                                    }))
+                            .onSuccess(body -> {
+                                assertThat("Body response", body.toString().contains(GREETING));
+                                requestCheckpoint.flag();
+                            }).onFailure(throwable -> vertxTestContext.failNow(throwable.getMessage()));
+                });
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("HttpClient Vertx just with HTTP/2 protocol option set and default port")
+    void http2ProtocolTest(VertxTestContext vertxTestContext) {
+        HttpClientOptions httpClientOptions = new HttpClientOptions().setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(httpClientOptions).build();
+        httpClient.request(HttpMethod.GET, baseUri.getPort(), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> request.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> vertxTestContext.verify(() -> {
+                    assertThat("Body response", body.toString().contains(GREETING));
+                    vertxTestContext.completeNow();
+                })).onFailure(Throwable::printStackTrace);
+
+    }
+
+    @Test
+    @DisplayName("Test max header size exceeded error message")
+    @Order(5)
+    void testMaxHeaderSizeExceeded(VertxTestContext testContext) {
+        HttpClientOptions options = new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustAll(true)
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(options).build();
+
+        int expectedLimit = 2048;
+
+        for (int i = 0; i < expectedLimit; i++) {
+            AtomicInteger headerCount = new AtomicInteger(i);
+
+            try {
+                request(headerCount, testContext);
+            } catch (Exception e) {
```

Comment:
How can this ever happen that you can catch Exception here? I would expect `onFailure` be called otherwise it is a bug?